sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -115,6 +116,69 @@ abstract class OrcSuite extends OrcTest with BeforeAndAfterAll {
     }
   }
 
+  protected def testSelectiveDictionaryEncoding(isSelective: Boolean) {
+    val tableName = "orcTable"
+
+    withTempDir { dir =>
+      withTable(tableName) {
+        val sqlStatement = orcImp match {
+          case "native" =>
+            s"""
+               |CREATE TABLE $tableName (zipcode STRING, uniqColumn STRING, value DOUBLE)
+               |USING ORC
+               |OPTIONS (
+               |  path '${dir.toURI}',
+               |  orc.dictionary.key.threshold '1.0',
+               |  orc.column.encoding.direct 'uniqColumn'
```

Comment:
This new feature needs a doc update. We need to let our end users how to use it.

Reply:
Ur, Apache ORC is an independent Apache project which has its own website and documents. We should respect that. If we introduce new ORC configuration one by one in Apache Spark website, it will eventually duplicate Apache ORC document in Apache Spark document.

We had better guide ORC fans to Apache ORC website. If something is missing there, they can file an ORC JIRA, not SPARK JIRA.

Reply:
I am fine either way. However, our current doc does not explain we are passing the data source specific options to the underlying data source:

https://spark.apache.org/docs/latest/sql-programming-guide.html#manually-specifying-options

Could you help improve it?

Reply:
Also give an example?

Reply:
That sounds like a different issue. This PR covers both `TBLPROPERTIES` and `OPTIONS` syntaxes where are designed for that configuration-purpose historically. I mean this is not about data-source specific PR. Also, the scope of this PR is only write-side configurations.

In any way, +1 for adding some introduction section for both Parquet/ORC examples there. We had better give both read/write side configuration examples, too. Could you file a JIRA issue for that?

Reply:
Maybe, dictionary encoding could be a good candidate; `parquet.enable.dictionary` and `orc.dictionary.key.threshold` et al.

Reply:
https://issues.apache.org/jira/browse/SPARK-25656 is created for that.